src/setup.jl

```diff
@@ -0,0 +1,109 @@
+
+function __init__()
+    Libdl.dlopen(rtlib, RTLD_GLOBAL)
```

Comment:
```suggestion
    include(joinpath(pathof(:mk), "deps", "deps.jl"))
    
     if Sys.iswindows()
            # On Windows, we have to open the threading library before calling libmkl_rt
            dlopen(libmkl_intel_thread)
     end
    Libdl.dlopen(rtlib, RTLD_GLOBAL)
```

Reply:
This likely won't work if someone uses standalone MKL instead of MKL.jl, and therefore doesn't have MKL.jl installed.

Reply:
I cannot get the correct path to deps.jl of MKL

Reply:
> 
> 
> This likely won't work if someone uses standalone MKL instead of MKL.jl, and therefore doesn't have MKL.jl installed.

We need to get the path of `libmkl_intel_thread` somehow. Either user should provide it manually, or we should use this automatic method for MKL.jl

Reply:
Based on my recent experience the way to go is probably to add the folder to the search path of dlopen. 
On unix installing MKL either via MKL.jl or standalone sorts that out. Currently on it.

Reply:
It appears what needs to be changed is just the name of the libary for Windows. 
```
julia> using Libdl
julia> Libdl.dlpath(:mkl_rt)                                                                                  
"C:\\Users\\Steffen\\.julia\\packages\\MKL\\goLVK\\deps\\usr\\bin\\mkl_core.dll"
```
Making changes now.

Reply:
Oh yes. That works! Great

Reply:
```julia
dlopen(Libdl.dlpath(:mkl_rt)  )
```
will work

Reply:
> 
> 
> It appears what needs to be changed is just the name of the libary for Windows.
> 
> ```
> julia> using Libdl
> julia> Libdl.dlpath(:mkl_core)                                                                                  
> "C:\\Users\\Steffen\\.julia\\packages\\MKL\\goLVK\\deps\\usr\\bin\\mkl_core.dll"
> ```
> 
> Making changes now.

Not the library name, but we need to use an explicit path. 

```julia
dlopen(Libdl.dlpath(:mkl_rt))
```
will work

Reply:
no need for the `dlpath`, `Libgen.dlopen(:mkl_rt)` works directly.

Reply:
> 
> 
> no need for the `dlpath`, `Libgen.dlopen(:mkl_rt)` works directly.
Oh that is true. On windows it is `:mkl_rt` compared to `:libmkl_rt`

Reply:
just updated the branch, it works on my machine, please test.

Reply:
This should be marked as resolved